pkg/ingress/controller/controller.go

```diff
@@ -824,10 +824,17 @@ func (c *Controller) ensureIngress(ing *nwv1.Ingress) error {
 			var policyRules []l7policies.CreateRuleOpts
 
 			if host != "" {
+
+				// Escape dot to be regex proof
+				hostRegex := strings.ReplaceAll(host, ".", "\\.")
```

Comment:
I realise you've just moved this here, but if the goal is to escape a string which will later be interpreted by a regexp parser (which one, btw?) then this isn't it.

Reply:
I'm not sure I follow, this was the existing "escaping" in place already and I suppose it was that simple because the `ingress` resource, sanitize its host field via API constraints, more precisely:
> a wildcard DNS-1123 subdomain must start with '*.', followed by a valid DNS subdomain, which must consist of lower case alphanumeric characters, '-' or '.' and end with an alphanumeric character (e.g. '*.example.com', regex used for validation is '\*\.[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

If I only consider wildcard at the beginning of the host` string and sanitize the rest with something like https://pkg.go.dev/regexp#QuoteMeta, would that be acceptable?

Reply:
Should have been clearer in my comment: I know you didn't add this line; you just moved it, so this isn't a problem introduced by this PR.

It did make me notice it, though, which is why I commented!